docs(#1036): Update linux installation instructions via apt

Nikolai2038 opened the pull request:
# Description

Fix #1036 - I modify instructions in READMEs to avoid two errors when installing Bruno in Linux via Apt (see issue for more info).

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

Its-treason commented on docs/readme/readme_bn.md:
```diff
@@ -41,12 +41,11 @@ choco install bruno
 echo "deb [signed-by=/etc/apt/keyrings/bruno.gpg] http://debian.usebruno.com/ bruno stable" | sudo tee /etc/apt/sources.list.d/bruno.list
-
 sudo apt update
+sudo apt install libasound2
```
Instead of adding the package to the docs, we can just add it as a dependency to the package in the [build config](https://github.com/usebruno/bruno/blob/main/packages/bruno-electron/electron-builder-config.js): https://www.electron.build/configuration/linux#debian-package-options

Nikolai2038 replied:
I don't have enough knowledge myself to implement this, but idea is right - hope someone can add this 🙏

Its-treason replied:
I just tested it locally, adding this into https://github.com/usebruno/bruno/blob/main/packages/bruno-electron/electron-builder-config.js after line 36 works. Can you add this to you're PR?
```js
  deb: {
    // Docs: https://www.electron.build/configuration/linux#debian-package-options
    depends: [
      "libgtk-3-0",
      "libnotify4",
      "libnss3",
      "libxss1",
      "libxtst6",
      "xdg-utils",
      "libatspi2.0-0",
      "libuuid1",
      "libsecret-1-0",
      "libasound2" // #1036
    ],
  },
```

```
❯ dpkg -I packages/bruno-electron/out/bruno_1.2.0_amd64_linux.deb
 new Debian package, version 2.0.
 size 69392000 bytes: control archive=3055 bytes.
     474 bytes,    14 lines      control
    5236 bytes,    84 lines      md5sums
     265 bytes,    10 lines   *  postinst             #!/bin/bash
      68 bytes,     4 lines   *  postrm               #!/bin/bash
 Package: bruno
 Version: 1.2.0
 License: unknown
 Vendor: Anoop M D <anoop.md1421@gmail.com>
 Architecture: amd64
 Maintainer: Anoop M D <anoop.md1421@gmail.com>
 Installed-Size: 272070
 Depends: libgtk-3-0, libnotify4, libnss3, libxss1, libxtst6, xdg-utils, libatspi2.0-0, libuuid1, libsecret-1-0, libasound2
 Recommends: libappindicator3-1
 Section: default
 Priority: optional
 Homepage: https://www.usebruno.com
 Description:
   Opensource API Client for Exploring and Testing APIs
```

Nikolai2038 replied:
Sure, I pushed new commits!

But I wasn't able to reproduce package generation - when I run:

```bash
npm run dist:deb
```

I got:

```
...
> bruno@v1.2.0 dist:deb
> electron-builder --linux deb --config electron-builder-config.js

  • electron-builder  version=23.0.2 os=10.0.22631
  • loaded configuration  file=E:\Forks\bruno\packages\bruno-electron\electron-builder-config.js
  • writing effective config  file=out\builder-effective-config.yaml
  • packaging       platform=linux arch=x64 electron=21.1.1 appOutDir=out\linux-unpacked
  • scheduling remote build  target=deb arch=x64
  • cannot get, wait  error=Get "https://service.electron.build/find-build-agent?no-cache=1ilvbkt": dial tcp 51.15.76.176:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
                      attempt=0
                      waitTime=2
  • cannot get, wait  error=Get "https://service.electron.build/find-build-agent?no-cache=1ilvbkt": dial tcp 51.15.76.176:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
                      attempt=1
                      waitTime=4
  • cannot get, wait  error=Get "https://service.electron.build/find-build-agent?no-cache=1ilvbkt": dial tcp 51.15.76.176:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
                      attempt=2
                      waitTime=6
  ⨯ Get "https://service.electron.build/find-build-agent?no-cache=1ilvbkt": dial tcp 51.15.76.176:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
  ⨯ E:\Forks\bruno\node_modules\app-builder-bin\win\x64\app-builder.exe process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
Exit code:
1  failedTask=build stackTrace=Error: E:\Forks\bruno\node_modules\app-builder-bin\win\x64\app-builder.exe process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
...
```

I am not familiar with Electron, but I will try to investigate it later...

Nikolai2038 replied:
Okay, I successfully overcome this by using docker container as mentioned [here](https://github.com/electron-userland/electron-build-service/issues/9).

Nikolai2038 commented:
- Added instruction to install `gpg` (I tried on new Debian WSL, and it is not installed by default).
- Fixed instruction to create `/root/.gnupg` directory - with correct rights to avoid this warning:

  ```
  gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
  ```

antony commented:
These instructions worked for me after a lot of searching online. Mostly, the `sudo mkdir -m 700 -p /root/.gnupg`. Not sure why this is needed, but it is.

janos-r commented:
This is definitely not necessary on LinuxMint and should not be added as a general rule.
To add anything manually into /root/ just to satisfy missing configuration to install something sounds crazy!
Find out the cause and what is the standard fix.
Is this happening only on debian?
Is it happening only if you don't have previously gpg installed?

antony commented:
My gpg was configured and already in use.

I searched comprehensively for a fix, to no avail.

Only creating this thing in root fixed it - I'm using Pop_Os

janos-r commented:
Ok, I tried on a fresh linuxMint install in a VM and this is my result:
gpg is preinstalled, but if you never ran it before, than it didn't yet create the .gnupg folder in either /root/ or ~/
I don't know why gpg doesn't create it automatically when importing this key, but rather than creating the folder manually, it is better to let gpg create it. It also creates your own public key and other files that belong there! As explained in https://www.redhat.com/sysadmin/getting-started-gpg

So ok, I agree this check can be part of the install script for linux. But instead of doing it with mkdir, it can be checked with just `sudo gpg --list-keys`. It will either list the subscribed public keys under root (there shouldn't be any, normally you don't need sudo to verify files) or it will create the folder with everything that belongs there:

```
$ gpg --list-keys
gpg: directory '/home/bestuser/.gnupg' created
gpg: keybox '/home/bestuser/.gnupg/pubring.kbx' created
gpg: /home/bestuser/.gnupg/trustdb.gpg: trustdb created
```

kinuax commented:
I confirm `sudo gpg --list-keys` creates the needed `/root/.gnupg' directory, thanks!

kinuax commented on docs/contributing/contributing_pl.md:
`````diff
@@ -33,7 +33,7 @@ Bruno jest rozwijane jako aplikacja desktopowa. Musisz załadować aplikację, u
 ### Lokalny Rozwój
 
-```bash
+````bash
`````
I think there is extra `.

kinuax commented on docs/contributing/contributing_pl.md:
`````diff
@@ -66,7 +66,7 @@ done
 find . -type f -name "package-lock.json" -delete
 
-```
+````
`````
Same here.

Nikolai2038 replied:
Thanks! Didn't notice that after merge. I used merge in WebStorm. It seems, that it was modified by git hooks, which formatted this braces due to lack of `` ``` `` for `` ```bash ``. I think it tried to convert:

````
```bash
```shell
```
````

to

`````
````bash
```shell
````
`````

I added missed `` ``` `` and changed `` ```` `` back to `` ``` ``.

Nikolai2038 replied:
It looks like someone already pushed this typo in main branch - you can see `` ```` `` in this [file](https://github.com/usebruno/bruno/blob/main/docs/contributing/contributing_pl.md?plain=1). In this PR I squished some commits, so I did not include this fix because it is not related to the changes I made

nielsbom commented on readme.md:
```diff
@@ -61,12 +61,11 @@ flatpak install com.usebruno.Bruno
 
 # On Linux via Apt
 sudo mkdir -p /etc/apt/keyrings
+sudo apt-get update && sudo apt-get install gpg
```
I believe `apt` should be used for end-users and `apt` for scripts.

[Source](https://askubuntu.com/questions/445384/what-is-the-difference-between-apt-and-apt-get)

nielsbom commented on readme.md:
```diff
@@ -61,12 +61,11 @@ flatpak install com.usebruno.Bruno
 # On Linux via Apt
 sudo mkdir -p /etc/apt/keyrings
+sudo apt-get update && sudo apt-get install gpg
+sudo gpg --list-keys
```
I think it probably is too noisy to add a comment here for _why_ this is needed. But maybe squashing the commits and adding the necessity for this line there?

Nikolai2038 replied:
You mean, something like that?

```bash
sudo gpg --list-keys && sudo gpg --no-default-keyring --keyring ...
```

nielsbom replied:
The command to list keys looks like a noop, but it actually does something (as discussed in the PR). This is surprising, so a comment may help.

But: a comment pointing out _why_ this command is there, _at that place_ is probably not relevant for 99% of users. So that's why I suggested squashing the commits and adding a line about this to the commit message.

Nikolai2038 replied:
The [commit](https://github.com/usebruno/bruno/pull/1037/commits/0809d3f69821bb5ac644b3bb2b78616fc452f374), where I added this command, has message:

```
In APT instructions: Use "gpg --list-keys" to create ".gnupg" directory
```

Are you sure squashing needed?

nielsbom replied:
Ah, didn't see that one, my bad.

I think in general squashing commits from (esp smaller) feature branches is a good thing. But it depends a bit on the practices of the project.

Nikolai2038 replied:
Alright, I've squished commits related to gpg now - you can see [commit](https://github.com/Nikolai2038/bruno/commit/b8a51a81261e78d7dfe6ae3462f0e717c7391e5f)

maintainer-bruno replied:
@Nikolai2038 @nielsbom Thanks for taking care of this.

At Bruno, we recommend squashing commits that are isolated to small, independent features or bug fixes. This helps keep the commit history clean and focused.

nielsbom commented on readme.md:
```diff
@@ -61,12 +61,11 @@ flatpak install com.usebruno.Bruno
-
-sudo apt update
-sudo apt install bruno
+sudo apt-get update && sudo apt-get install bruno
```
I believe `apt` should be used for end-users and `apt` for scripts.

[Source](https://askubuntu.com/questions/445384/what-is-the-difference-between-apt-and-apt-get)

Nikolai2038 replied:
Thanks for pointing out! I didn't know that - I was working with distributions without `apt` installed by default, so I wrote `apt-get` here. I changed it back to `apt` in recent commit.

helloanoop commented:
@Nikolai2038 

All looks good. I would need to review this part to make sure that all of these are really needed. And that will take time.
```js
deb: {
    // Docs: https://www.electron.build/configuration/linux#debian-package-options
    depends: [
      'libgtk-3-0',
      'libnotify4',
      'libnss3',
      'libxss1',
      'libxtst6',
      'xdg-utils',
      'libatspi2.0-0',
      'libuuid1',
      'libsecret-1-0',
      'libasound2' // #1036
    ]
  },
```

Can you submit the above part as a separate PR and keep this PR just to update the linux install instructions ?

Nikolai2038 commented:
@helloanoop Thanks for the response! I updated this branch and created separate [pull request](https://github.com/usebruno/bruno/pull/2356).

gioisco commented:
I was trying to install Bruno on my "Pop!_OS 22.04 LTS" from the repository and encountered an issue because the `/root/.gnupg` directory did not exist.

Here the error I received:

```bash
$ sudo gpg --no-default-keyring --keyring /etc/apt/keyrings/bruno.gpg --keyserver keyserver.ubuntu.com --recv-keys 9FA6017ECABE0266
gpg: keybox '/etc/apt/keyrings/bruno.gpg' created
gpg: failed to create temporary file '/root/.gnupg/.#lk0x0000589d4cd992d0.pop-os.24118': No such file or directory
gpg: connecting dirmngr at '/root/.gnupg/S.dirmngr' failed: No such file or directory
gpg: keyserver receive failed: No dirmngr
````

I solved with:
```bash
$ sudo gpg --list-keys
gpg: directory '/root/.gnupg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
```

I am writing this comment because once this pull request is merged, the instructions at http://debian.usebruno.com/ should also be updated accordingly.

JoaoGesser commented:
I was trying to install Bruno via apt but didnt work but with this approach it works. Thank you for this contribution! @Nikolai2038

ramki-bruno commented on docs/readme/readme_pl.md:
```diff
@@ -69,12 +69,11 @@ flatpak install com.usebruno.Bruno
+sudo gpg --list-keys
 sudo gpg --no-default-keyring --keyring /etc/apt/keyrings/bruno.gpg --keyserver keyserver.ubuntu.com --recv-keys 9FA6017ECABE0266
-
 echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/bruno.gpg] http://debian.usebruno.com/ bruno stable" | sudo tee /etc/apt/sources.list.d/bruno.list
```
Looks like this line got updated with `deb [arch=amd64 signed...` [here already](https://github.com/usebruno/bruno/pull/2392/files?diff=unified&w=0#diff-ce94be584b5ab653b8699c9597c768a45f0771933ddd932a43f400c2e082c997R73) sometime ago which caused the conflict.
